plugins/catalog-backend/src/ingestion/processors/LocationRefProcessor.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity, LocationRef, LocationSpec } from '@backstage/catalog-model';
+import * as result from './results';
+import { LocationProcessor, LocationProcessorEmit } from './types';
+
+export class LocationRefProcessor implements LocationProcessor {
+  async processEntity(
+    entity: Entity,
+    _location: LocationSpec,
+    emit: LocationProcessorEmit,
+  ): Promise<Entity> {
+    if (entity.kind === 'LocationRef') {
+      const location = entity as LocationRef;
+      if (location.spec.target) {
+        emit(
+          result.location(
+            { type: location.spec.type, target: location.spec.target },
```

Comment:
Could just go location.spec here tbh, since we're in TypeScript it can't really be abused

Reply:
I didn't want to leak additional fields into it - it's just confusing when debugging to have junk in there. This way I'm certain what gets transferred especially if the format evolves later. Would like to keep as is

Reply:
Yeah keep it, typed out this comment before I realized there was a `targets` as well, so it makes sense.